ADAL/src/ADLogger.m

```diff
@@ -329,9 +353,14 @@ + (void)logToken:(NSString *)token
 + (void)setIdValue:(NSString*)value
             forKey:(NSString*)key
 {
-    [self adalId];
+    [self adalMetadata];
     
-    [s_adalId setObject:value forKey:key];
+    [s_adalFullMetadata setObject:value forKey:key];
+
+    if (s_adalShortMetadata[key])
+    {
+        [s_adalShortMetadata setObject:value forKey:key];
+    }
```

Comment:
I am still thinking, should broker use this API to set the app name and app version? if we do, we may need an API to reset adalMetadata

Reply:
What is broker using now for adal version?

Reply:
Broker relies on ADAL to set the adal version, by calling [`ADAuthenticationRequest acquireToken:]`,  [`ADAuthenticationRequest requestCode:]`and `[ADWebAuthRequest send:].

For app name/version, seems broker requires them  to be context/request specific. I am thinking one possible solution is to 
- have readonly properties for app name/version in `ADAuthentionRequest`
- have an overload [ADLogger adalShortMetadata:(NSDictionary *)overwriteDefaultMetadata]
Any other idea? :)